src/Beman/Example/CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Add the library.
+add_library(beman_example STATIC "")
+add_library(beman::example ALIAS beman_example)
+
+# Using the full project version because example does not provide a stable ABI
+# between shared library versions
+set_property(TARGET beman_example PROPERTY SOVERSION "${PROJECT_VERSION}")
+
+# Set the C++ sources.
+target_sources(beman_example PRIVATE
+  example.cpp
+)
```

Comment:
We should also be adding the headers here (as part of a FILE_SET) so they show up as part of the beman_example target in IDEs and are installed automatically as part of `make install`. I see something like that below, but it is commented out.